### ISSUE TITLE
Log stock actions for auditing

### ISSUE THREAD

**Issue by tombola** [vitalseeds/vs-data-api#6]

**Comment by tombola** [comment]
**tl;dr**  First instance custom text based logger for INFO level only, log WC and VS persistence actions.

- 💡 could do this with a custom logger (action_audit?)
    - some variation on actor/verb/object
        - allow for callback to determine actor
        - in our case actor may just be `module.function`
- 💡 optionally save to csv for viewing in rich/spreadsheet?
    - _optionally save to sqlite?_
- 💡 save parameters/response as json/parquet file, associated by id
    - textualise app to view/filter logs and display related json
    - save json to sqlite by unique log id instead of file?
    - _allow for undo from log information?.._
- delete older logs on startup
- package with flit?
    - pypi candidate


See pypi for package examples, eg `audit-log`.

## Resources

- https://towardsdatascience.com/8-advanced-python-logging-features-that-you-shouldnt-miss-a68a5ef1b62d
- https://flit.pypa.io/en/latest/pyproject_toml.html

**Comment by tombola** [comment]
_Possibly_ could catch **all** errors during fastapi request in a try/catch (and return error message to FM via json), whilst still outputting full trace to logs:

https://realpython.com/python-logging/#capturing-stack-traces

**Comment by tombola** [comment]
Also separately log all SQL queries.
